src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2294,7 +2310,22 @@ public ConstructorEdit(INamedTypeSymbol oldType)
                                         continue;
                                     }
 
-                                    if (!newSymbol.IsImplicitlyDeclared)
+                                    if (IsRecordPrimaryConstructorProperty(edit.OldNode))
+                                    {
+                                        // When deleting a property of a record that represents a primary constructor parameter we need to mark each
+                                        // part of the property as processed so that they don't become delete edits. This isn't an issue for normal
+                                        // properties because adding them is a rude edit so it never gets this far.
+                                        var oldProperty = (IPropertySymbol)oldSymbol;
+                                        if (oldProperty.GetMethod is not null)
+                                        {
+                                            processedSymbols.Add(oldProperty.GetMethod);
+                                        }
+                                        if (oldProperty.SetMethod is not null)
+                                        {
+                                            processedSymbols.Add(oldProperty.SetMethod);
+                                        }
+                                    }
+                                    else if (!newSymbol.IsImplicitlyDeclared)
```

Comment:
>else [](start = 36, length = 4)

Isn't this case valid for properties in records as well that have been moved to another partial declaration?

Reply:
Yes, well spotted. Right now moving an explicit property of a record from one partial to another will report an unnecessary constructor edit. Will have to see how I can prevent that.

Reply:
Didn't need to go that far, but this also showed up some bugs around explicitly implementing properties with bodies, so have changed this a bunch. I think its neater now (hopefully).